src/dmd/cppmanglewin.d

```diff
@@ -202,27 +202,26 @@ public:
         case Tfloat64:
             buf.writeByte('N');
             break;
-        case Tbool:
-            buf.writestring("_N");
-            break;
-        case Tchar:
-            buf.writeByte('D');
-            break;
-        case Tdchar:
-            buf.writeByte('I');
-            break;
-            // unsigned int
         case Tfloat80:
             if (flags & IS_DMC)
                 buf.writestring("_Z"); // DigitalMars long double
             else
                 buf.writestring("_T"); // Intel long double
             break;
+        case Tbool:
+            buf.writestring("_N");
+            break;
+        case Tchar:
+            buf.writeByte('D');
+            break;
         case Twchar:
             if (flags & IS_DMC)
                 buf.writestring("_Y"); // DigitalMars wchar_t
             else
-                buf.writestring("_W"); // Visual C++ wchar_t
```

Comment:
This a breaking change.

Reply:
It's a *fixing* change. It's already broken.

Reply:
Nobody can depend on the extern(C++) string stuff, because it's totally broken! We need to finally straighten this out.

`wchar_t` needs a typedef, like `c_long`.